packages/driver/src/cy/net-stubbing/events/before-request.ts

```diff
@@ -119,8 +119,60 @@ export const onBeforeRequest: HandlerFn<CyHttpMessages.IncomingRequest> = (Cypre
   let resolved = false
   let handlerCompleted = false
 
+  const createQueryObject = () => {
+    try {
+      const url = new URL(req.url)
+      const urlSearchParams = new URLSearchParams(url.search)
+      const result = {}
+
+      for (let pair of urlSearchParams.entries()) {
+        result[pair[0]] = pair[1]
+      }
+
+      return result
+    } catch { // invalid url
+      return {}
+    }
+  }
+
+  const updateUrlParams = (paramsObj) => {
+    const url = new URL(req.url)
+    const urlSearchParams = new URLSearchParams(paramsObj)
+
+    url.search = urlSearchParams.toString()
+    req.url = url.toString()
+  }
+
+  const createQueryProxy = (obj) => {
+    return new Proxy(obj, {
+      set (target, key, value) {
```

Comment:
This `Proxy` is incomplete, I think, it needs to implement `deleteProperty` and throw or handle `defineProperty` and `setPrototypeOf` to prevent the `obj` from becoming out of sync with `target`.

List of handler functions - the rest are read-only I think, so shouldn't matter: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy#handler_functions

Reply:
Done.

Reply:
@sainthkh it still does not handle `setPrototypeOf` or `defineProperty` - it's ok if these just throw, but I think they need to exist to block users from getting obj out of sync with `target`.